WordPress/gutenberg
# Image: Fix design of close button for the lightbox

## What?

This PR changes the design of the close button in the lightbox so that it is easily visible at all times.

## Why?

Fixes #51126, wherein the close button was appearing on top of images and was hard to see in cases where the image was dark.

## How?

It rebases on top of #51164 to make use of those fixes that ensures images display properly.

It uses [this design](https://github.com/WordPress/gutenberg/issues/51126#issuecomment-1573835454) and sets the close button color to be the theme contrast color, adds padding, and makes the close button smaller.

## Testing Instructions




1. Ensure the Interactivity API and Behaviors UI setting is enabled in Gutenberg > Experim1.ents in the admin.
2. Create a new post and insert a vertically oriented image as well as a horizontally oriented one.
3. Activate the image lightboxes via Advanced > Behaviors > Lightbox in block settings.
4. Publish and view the post.
5. Click on the images; see that the image is fully visible and contained within the lightbox; the close button should be clearly visible and unobscured by the image.
6. Change the theme background color and contrast color. (Note: Manually setting these colors individually will not work, I believe due to [this issue](https://github.com/WordPress/gutenberg/issues/49712), but you can experiment with changing theme styles.)
7. Test again to ensure that the colors are applied as expected.

### Testing Instructions for Keyboard

- When viewing the post, tab to the image instead and press Enter or Space to open the lightbox.
- See that the close button and focus are visible.
